iroha_logger/src/layer.rs

```diff
@@ -0,0 +1,122 @@
+    subscriber::Interest,
+    Event, Id, Level, Metadata, Subscriber,
+};
+use tracing_core::span::Current;
+
+/// Trait for filtering or inspecting events
+pub trait EventFilterTrait: 'static {
+    /// Inner subscriber of current layer
+    type Subscriber: Subscriber;
+    /// Function which filters or inspects events
+    fn event(&self, event: &Event<'_>);
+    /// Basically deref for inner subscriber
+    fn inner_subscriber(&self) -> &Self::Subscriber;
+}
+
+/// Wrapper which implements `Subscriber` trait for any implementator of `EventfilterTrait`
+#[derive(Debug, Clone)]
+pub struct EventFilter<E: EventFilterTrait>(pub E);
+
+impl<E: EventFilterTrait> Subscriber for EventFilter<E> {
+    fn enabled(&self, metadata: &Metadata<'_>) -> bool {
+        self.0.inner_subscriber().enabled(metadata)
+    }
+    fn new_span(&self, span: &Attributes<'_>) -> Id {
+        self.0.inner_subscriber().new_span(span)
+    }
+    fn record(&self, span: &Id, values: &Record<'_>) {
+        self.0.inner_subscriber().record(span, values)
+    }
+    fn record_follows_from(&self, span: &Id, follows: &Id) {
+        self.0.inner_subscriber().record_follows_from(span, follows)
+    }
+    fn enter(&self, span: &Id) {
+        self.0.inner_subscriber().enter(span)
+    }
+    fn exit(&self, span: &Id) {
+        self.0.inner_subscriber().exit(span)
+    }
+    fn register_callsite(&self, metadata: &'static Metadata<'static>) -> Interest {
+        self.0.inner_subscriber().register_callsite(metadata)
+    }
+    fn max_level_hint(&self) -> Option<TracingLevelFilter> {
+        self.0.inner_subscriber().max_level_hint()
+    }
+    fn clone_span(&self, id: &Id) -> Id {
+        self.0.inner_subscriber().clone_span(id)
+    }
+    #[allow(deprecated)]
+    fn drop_span(&self, id: Id) {
+        self.0.inner_subscriber().drop_span(id)
+    }
+    fn try_close(&self, id: Id) -> bool {
+        self.0.inner_subscriber().try_close(id)
+    }
+    fn current_span(&self) -> Current {
+        self.0.inner_subscriber().current_span()
+    }
+    #[allow(unsafe_code)]
+    unsafe fn downcast_raw(&self, id: TypeId) -> Option<*const ()> {
+        self.0.inner_subscriber().downcast_raw(id)
+    }
+
+    fn event(&self, event: &Event<'_>) {
+        self.0.event(event)
+    }
+}
+
+/// Filter of logs by level
+#[derive(Debug)]
+pub struct LevelFilter<S: Subscriber> {
+    subscriber: S,
+}
+
+static CURRENT_LEVEL: AtomicU8 = AtomicU8::new(0);
+
+impl<S: Subscriber> LevelFilter<S> {
+    fn level_as_u8(level: Level) -> u8 {
+        match level {
+            Level::TRACE => 0,
+            Level::DEBUG => 1,
+            Level::INFO => 2,
+            Level::WARN => 3,
+            Level::ERROR => 4,
+        }
+    }
+
+    /// Constructor of level filter
+    #[allow(clippy::new_ret_no_self)]
+    pub fn new(level: Level, subscriber: S) -> impl Subscriber {
+        Self::update_max_level(level);
+        EventFilter(Self { subscriber })
+    }
```

Comment:
It is confusing, it doesn't create LevelFilter but rather EventFilter?

Reply:
Yes. Because we would rather need `EventFilter` because it implements Subscriber